Ws to tls proxy misinterprets begin extension

```
org.agrona.concurrent.AgentTerminationException: java.lang.NullPointerException: Cannot invoke "io.aklivity.zilla.runtime.binding.tcp.internal.types.ProxyAddressFamily.ordinal()" because the return value of "io.aklivity.zilla.runtime.binding.tcp.internal.types.ProxyAddressFW.kind()" is null
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:572)
	at org.agrona.core/org.agrona.concurrent.AgentRunner.doDutyCycle(AgentRunner.java:291)
	at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:164)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: Cannot invoke "io.aklivity.zilla.runtime.binding.tcp.internal.types.ProxyAddressFamily.ordinal()" because the return value of "io.aklivity.zilla.runtime.binding.tcp.internal.types.ProxyAddressFW.kind()" is null
	at io.aklivity.zilla.runtime.binding.tcp/io.aklivity.zilla.runtime.binding.tcp.internal.types.ProxyAddressFW.tryWrap(ProxyAddressFW.java:79)
	at io.aklivity.zilla.runtime.binding.tcp/io.aklivity.zilla.runtime.binding.tcp.internal.types.stream.ProxyBeginExFW.tryWrap(ProxyBeginExFW.java:52)
	at io.aklivity.zilla.runtime.binding.tcp/io.aklivity.zilla.runtime.binding.tcp.internal.types.OctetsFW.get(OctetsFW.java:15)
	at io.aklivity.zilla.runtime.binding.tcp/io.aklivity.zilla.runtime.binding.tcp.internal.stream.TcpClientFactory.newStream(TcpClientFactory.java:130)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleBeginInitial(DispatchAgent.java:1278)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleDefaultReadInitial(DispatchAgent.java:1070)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleReadInitial(DispatchAgent.java:1010)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleRead(DispatchAgent.java:953)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.concurent.ManyToOneRingBuffer.read(ManyToOneRingBuffer.java:181)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:566)
	... 3 more
	Suppressed: java.lang.Exception: [engine/data#9]	[0x0909000000000009] streams=[consumeAt=0x000005c8 (0x00000000000005c8), produceAt=0x000005c8 (0x00000000000005c8)]
		at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:570)
		... 3 more
```

--- a/runtime/binding-tcp/src/main/java/io/aklivity/zilla/runtime/binding/tcp/internal/stream/TcpClientFactory.java
+++ b/runtime/binding-tcp/src/main/java/io/aklivity/zilla/runtime/binding/tcp/internal/stream/TcpClientFactory.java
@@ -51,6 +51,7 @@ import io.aklivity.zilla.runtime.binding.tcp.internal.types.stream.AbortFW;
 import io.aklivity.zilla.runtime.binding.tcp.internal.types.stream.BeginFW;
 import io.aklivity.zilla.runtime.binding.tcp.internal.types.stream.DataFW;
 import io.aklivity.zilla.runtime.binding.tcp.internal.types.stream.EndFW;
+import io.aklivity.zilla.runtime.binding.tcp.internal.types.stream.ExtensionFW;
 import io.aklivity.zilla.runtime.binding.tcp.internal.types.stream.ProxyBeginExFW;
 import io.aklivity.zilla.runtime.binding.tcp.internal.types.stream.ResetFW;
 import io.aklivity.zilla.runtime.binding.tcp.internal.types.stream.WindowFW;
@@ -78,6 +79,7 @@ public class TcpClientFactory implements TcpStreamFactory
     private final ResetFW.Builder resetRW = new ResetFW.Builder();
     private final WindowFW.Builder windowRW = new WindowFW.Builder();
 
+    private final ExtensionFW extensionRO = new ExtensionFW();
     private final ProxyBeginExFW beginExRO = new ProxyBeginExFW();
     private final ProxyBeginExFW.Builder beginExRW = new ProxyBeginExFW.Builder();
 
@@ -126,8 +128,10 @@ public class TcpClientFactory implements TcpStreamFactory
         final BeginFW begin = beginRO.wrap(buffer, index, index + length);
         final long routeId = begin.routeId();
         final long authorization = begin.authorization();
-        final OctetsFW extension = begin.extension();
-        final ProxyBeginExFW beginEx = extension.get(beginExRO::tryWrap);
+        final ExtensionFW extension = begin.extension().get(extensionRO::tryWrap);
+        final ProxyBeginExFW beginEx = extension != null && extension.typeId() == proxyTypeId
+                ? begin.extension().get(beginExRO::tryWrap)
+                : null;
 
         InetSocketAddress route = null;
 

--- a/runtime/binding-tls/src/main/java/io/aklivity/zilla/runtime/binding/tls/internal/stream/TlsClientFactory.java
+++ b/runtime/binding-tls/src/main/java/io/aklivity/zilla/runtime/binding/tls/internal/stream/TlsClientFactory.java
@@ -892,6 +892,8 @@ public final class TlsClientFactory implements TlsStreamFactory
             final long traceId = begin.traceId();
             final long authorization = begin.authorization();
             final OctetsFW extension = begin.extension();
+            final ExtensionFW typedEx = extension.get(extensionRO::tryWrap);
+            final OctetsFW proxyEx = typedEx != null && typedEx.typeId() == proxyTypeId ? extension : EMPTY_OCTETS;
 
             assert acknowledge <= sequence;
             assert sequence >= initialSeq;
@@ -905,7 +907,7 @@ public final class TlsClientFactory implements TlsStreamFactory
 
             state = TlsState.openInitial(state);
 
-            client.doNetBegin(traceId, affinity, extension);
+            client.doNetBegin(traceId, affinity, proxyEx);
         }
 
         private void onAppFlush(

--- a/specs/binding-tls.spec/src/main/scripts/io/aklivity/zilla/specs/binding/tls/streams/network/connection.established.with.extension.data/client.rpt
+++ b/specs/binding-tls.spec/src/main/scripts/io/aklivity/zilla/specs/binding/tls/streams/network/connection.established.with.extension.data/client.rpt
@@ -18,8 +18,6 @@ connect "zilla://streams/net0"
   option zilla:window 65536
   option zilla:transmission "simplex"
 
-write zilla:begin.ext ${zilla:id("not-proxy")} "extension"
-
 connected
 
 write abort

--- a/specs/binding-tls.spec/src/main/scripts/io/aklivity/zilla/specs/binding/tls/streams/network/connection.established.with.extension.data/server.rpt
+++ b/specs/binding-tls.spec/src/main/scripts/io/aklivity/zilla/specs/binding/tls/streams/network/connection.established.with.extension.data/server.rpt
@@ -19,8 +19,6 @@ accept "zilla://streams/net0"
   option zilla:transmission "simplex"
 accepted
 
-read zilla:begin.ext ${zilla:id("not-proxy")} "extension"
-
 connected
 
 read aborted

